gstd/gstd_daemon.c

```diff
@@ -45,15 +47,23 @@ static gchar *gstd_daemon_get_pid_filename (const gchar * filename);
 static gboolean _initialized = FALSE;
 static gchar *pid_filename = NULL;
 
-void
+gboolean
 gstd_daemon_init (gint argc, gchar * argv[], gchar * pidfilename)
 {
   const gchar *process_name;
+  gchar *pid_path;
 
-  g_return_if_fail (argv);
+  g_return_val_if_fail (argv, FALSE);
 
   if (_initialized)
-    return;
+    goto out;
+
+  /* Check if pid dir available */
+  pid_path = gstd_daemon_get_pid_filename (pidfilename);
+  if (0 != g_access (pid_path, W_OK)) {
```

Comment:
Seems `pid_path` is not freed if the conditional is TRUE (0 != g_access (pid_path, W_OK))? 
or  `g_access` accepts the `NULLABLE` variables?

Reply:
g_access did handle the null pid_path, however, the message obtained for errno in this case was not very helpful ("Bad address"), better to catch this case separately.